src/features/dataset/DatasetHeader.tsx

```diff
@@ -31,9 +47,21 @@ const DatasetHeader: React.FC<DatasetHeaderProps> = ({
 }) => {
   const dispatch = useDispatch()
   const history = useHistory()
+  // const location = useLocation()
   const header = useSelector(selectDatasetHeader)
-  const loading = useSelector(selectIsHeaderLoading)
+  const dataset = useSelector(selectDataset)
+  const headerLoading = useSelector(selectIsHeaderLoading)
+  const datasetLoading = useSelector(selectIsDatasetLoading)
   const qriRef = qriRefFromVersionInfo(header)
+  const [ titleChangePosition, setTitleModalPosition ] = useState({ top: '0', left: '0' })
+
+  const measuredRef = useCallback(node => {
+    if (node !== null) {
+      const left: string = (node.getBoundingClientRect().left - 25).toString() + 'px'
+      const top: string = (node.getBoundingClientRect().top - 56).toString() + 'px'
```

Comment:
very cool!

Can I ask where the `25` & `56` come from? 

I'm asking because this works perfectly on chrome, but on safari and firefox the text jumps down a few pixels.

Reply:
I think it comes from paddings/ borders of the popup. @ramfox

Reply:
I let me re-phrase: did these numbers come from experimentation or do they come from knowing the pixels of the line height or something. Because, again, they work on chrome, but are different for safari and firefox

Reply:
I was experimenting

Reply:
I see the problem. Do you have any suggestions?

Reply:
This is how it calculates it on Chrome: 
left: 280.045
top: 118.864
This is on firefox: 
left: 178.5
top: 119.1

Reply:
thanks for doing the further research.

I've unfortunately realized we have a few more problems using this approach. Basically, this won't recalculate if you toggle the sidebar nav over, or if you make the browser window small & the smaller scrolling header folds down:
<img width="536" alt="Screen Shot 2021-11-10 at 9 04 57 PM" src="https://user-images.githubusercontent.com/17440142/141193766-b9b127cc-0ed7-473a-ad83-09a363609cf0.png">
(is that a bug?)

Since we know that the two potential changes to the position are when the sidebar goes in and out or the `DatasetMiniHeader` goes in or out, we can use the state and selectors to know whether they have changed from in to out or back, and we can recalculate then.

We already have the sidebar UI in the `App` state tree, so we can just `useSelector(selectNavExpanded)` to get `isNavExpanded`.

We can add a field `isMiniHeaderInView` (or some other name), to the `Dataset` state & add a `setIsMiniHeaderInView` action, that sets the field in the `Dataset` state, and create a `selectIsMiniHeaderInView` selector. 

Then, like we do for the `isNavExpanded`, we use `useSelector(selectIsMiniHeaderInView)` to get `isMiniHeaderInView` in `DatasetHeader`

Then, we just add `isNavExpanded` & `isMiniHeaderInView`  to the dependency array of the `useCallback` & that's one problem solved.

This doesn't fix the pixel differences between safari/firefox and chrome. I got side tracked when I discovered the above problem, lol.